src/components/templates/PosterHall/components/PosterPreview/PosterPreview.tsx

```diff
@@ -1,31 +1,66 @@
-import React, { useCallback, useMemo } from "react";
+import React, {
+  MouseEventHandler,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
 import classNames from "classnames";
 
-import { PosterPageVenue } from "types/venues";
+import { PersonalizedPoster, VenueEvent } from "types/venues";
 
 import { WithId } from "utils/id";
 
 import { PosterCategory } from "components/atoms/PosterCategory";
 
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faBookmark as solidBookmark } from "@fortawesome/free-solid-svg-icons";
+import { faBookmark as regularBookmark } from "@fortawesome/free-regular-svg-icons";
+
+import { updatePersonalizedSchedule, savePosterToProfile } from "api/profile";
+import { useUser } from "hooks/useUser";
+
+// import { UserAvatar } from "components/atoms/UserAvatar";
+// import { useProfileModalControls } from "hooks/useProfileModalControls";
+// import { useWorldUsersById } from "hooks/users";
+
+import { useRelatedVenues } from "hooks/useRelatedVenues";
+import { useVenueEvents } from "hooks/events";
+
+import { WithVenueId } from "utils/id";
+
 import "./PosterPreview.scss";
 
 export interface PosterPreviewProps {
-  posterVenue: WithId<PosterPageVenue>;
   enterVenue: (venueId: string) => void;
+  personalizedPoster: WithId<PersonalizedPoster>;
 }
 
+const emptyRelatedEvents: WithVenueId<VenueEvent>[] = [];
+export const emptyPersonalizedSchedule = {};
+
 export const PosterPreview: React.FC<PosterPreviewProps> = ({
```

Comment:
Keep `FooComponentProps` next to `FooComponent`.

---

Also, what's the need for the `emptyRelatedEvents` and `emptyPersonalizedSchedule` here? Particularly given this file is `PosterPreview` and seems to not relate to the schedule?

Reply:
This may indeed be an OHBM-specific issue, but the issue this PR is trying to address is that each poster is associated with three events. This is far too much to present visually on the schedule, but we want users to be able to bookmark a poster and to have the three events associated with 'saved' poster appear in their personalized schedule.

Also, fixed the props next to component issue

Reply:
_nods_ that makes sense. I expect it may end up being very OHBM-specific. But let's leave retargeting the branch till we know that for certain.

Reply:
removed during refactoring